### ISSUE TITLE
More than one cell per row: How to diff inside a section

### ISSUE THREAD

**Issue by yspreen** [Instagram/IGListKit#1474]
## New issue checklist

- [x] I have reviewed the [`README`](https://github.com/Instagram/IGListKit/blob/master/README.md) and [documentation](http://instagram.github.io/IGListKit)
- [x] I have searched [existing issues](https://github.com/Instagram/IGListKit/issues) and this is not a duplicate
- [ ] I have attempted to reproduce the issue and include an example project.

### General information

- `IGListKit` version: master
- iOS version(s): 14
- CocoaPods/Carthage version: 
- Xcode version:
- Devices/Simulators affected:
- Reproducible in the demo project? (Yes/No):
- Related issues:

So I've read numerous times that a section is generally only supposed to show one _object_. I'm struggling to understand how I would use IGLK to show a grid of items containing more than one cell per row?

If one section is always 100% of its container's width and cannot have multiple sections per row, how does the diffing on one section work? A section's data has to conform to ListDiffable. Meaning I have to provide the logic for section == section. But that's not in the spirit of using a library to do this diffing for me. I want to tell IGLK, 'here's a section; please diff all the cells in it.' But I really don't see how that's possible.

---

I assume `ListCollectionViewLayout` can be used for this purpose, but I couldn't find any documentation of how to use it properly. By just implementing `sizeForItem(at index: Int)` in the section controller, the cells still occupy an entire row. Can someone point in the right direction?

**Comment by lorixx** [comment]
Yes, it's better to 1 object for 1 section, but if your section contains multiple objects, then you might 1) build your own grid like cell for the multiple objects; or 2) Use customized `ListCollectionViewLayout` or UIColelctionViewLayout to customize the layouts. Feel free to check out one of the note I wrote before for the customized layout with a few options that might spark your work here: https://medium.com/@zhisheng_huang/messenger-seen-heads-animation-on-ios-1b06b7e610df

Basically, for 1) your data model would be an array of section objects; and for 2) your data model would be an array of row items, which you normalize each row into the array.
